angularjs-portal-home/src/main/webapp/my-app/layout/widget/partials/weather.html

```diff
@@ -1,23 +1,23 @@
 <div>
 
-    <div ng-if="!weather.location">
-		<div ng-model="noLocations" ng-init="noLocations=true">
-			<i class="fa fa-exclamation-triangle" aria-label='Warning'></i>
-			<div class="warning-message-weather-widget">Hold up! You haven't set any weather locations yet! Launch the full app to select your weather locations.</div>
-		</div>
-		
-		<div ng-if="!noLocations"id="loading">
-    		<loading-gif data-object='weatherData'></loading-gif>
-    	</div>
-	</div>
+    <div ng-if="weatherData && weatherData.length == 0 && !loading && !error">
+        <div>
+            <i class="fa fa-exclamation-triangle" aria-label='Warning'></i>
+            <div class="warning-message-weather-widget">Hold up! You haven't set any weather locations yet! Launch the full app to select your weather locations.</div>
```

Comment:
This seems just fine for now.  In time it may be worth getting to a more smart-widget-general shared `class` for styling error messages.  i.e., Smart Widget failure messages should probably look similar.